Clean updated Docker topics for Docker-1.0.0 (#6406)
magento/devdocs: src/cloud/docker/docker-containers.md

  - Configuration
---

The [Magento Cloud Docker repository][docker-repo] contains build information to create a Docker environment with the required specifications for Magento Cloud. All customized Docker containers are available in the repository. You can customize these containers and add more as needed.

> {{ site.data.var.mcd }} generates the `docker-compose.yml` file to the required specifications. Then, you use docker-compose to create the the container instances, build, and deploy the {{site.data.var.ee}} site.

Author: ```suggestion
Magento Cloud Docker generates the `docker-compose.yml` file to the required specifications. Then, you use docker-compose to create the the container instances, build, and deploy the {{site.data.var.ee}} site.
```

| ------------- | ---------- | ---------- | ------------------ |------------------
| [build] | Build Container |   |   | PHP Container, runs build process
| [cron]| Cron Jobs |   |   |  PHP Container, runs cron tasks
| [deploy] | Deploy Container |   |  |  PHP Container, runs the deploy process
| [db] | MariaDB     | `--db` | 10.0, 10.1, 10.2 |  Standard database container
| [elasticsearch] | Elasticsearch | `--es` | 1.7, 2.4, 5.2, 6.5 |
| [fpm][fpm-container] | PHP FPM | `--php` | 7.0, 7.1, 7.2, 7.3 |  Used for all incoming requests
| [node][node-container] | Node | `--node` | 6, 8, 10, 11 |  Used gulp or other NPM based commands
| [rabbitmq][rabbitmq-container]| RabbitMQ | `--rmq` | 3.5, 3.7, 3.8 |
| [redis][redis-container] | Redis     | `--redis` | 3.2, 4.0, 5.0 |   Standard redis container
| [selenium][selenium-container]| Selenium | `--with-selenium`<br>`--selenium-version`<br>`--selenium-image`| Any | Enables Magento application testing using the Magento Functional Testing Framework (MFTF)
| [tls][tls-container] | SSL Endpoint |  |   |  Terminates SSL, can be configured to pass to varnish or nginx
| [varnish][varnish-container] | Varnish | `--varnish` | 4, 6 |

> | [web][web-container] | Nginx | `--nginx` | 1.9, latest |

Reviewer: NGINX?

> | [web][web-container] | Nginx | `--nginx` | 1.9, latest |

Author: ```suggestion
| [web][web-container] | NGNIX | `--nginx` | 1.9, latest |
```

You do not interact with most of these volumes, which are used by the Docker containers and follow the docker-compose lifecycle. The only exception to this is the `magento-sync` directory which is an external volume used by the Mutagen application to transport data into the containers from the host operating system.

### Rebuild a clean environment

Use the following command to remove all data and rebuild a clean environment:

```bash
 bin/docker down
```

This command runs the `docker-compose down -v` command which removes all components of your local Docker instance including containers, networks, volumes, and images. This command does not [the persistent database volume][db] or the `magento-sync` volume used for file synchronization.

> The `magento-sync` volume is an external volume that you have to create or delete manually.

Author: ```suggestion
```

Reviewer: May be better to use "must" or something. "have to" is a bit clunky.
